src/raft.rs

```diff
@@ -2125,31 +2133,35 @@ impl<T: Storage> Raft<T> {
         let mut to_send = Message::default();
         to_send.to = m.from;
         to_send.set_msg_type(MessageType::MsgAppendResponse);
-        match self
+
+        if let Some(append_ents) = self
             .raft_log
             .maybe_append(m.index, m.log_term, m.commit, &m.entries)
         {
-            Some(mlast_index) => {
-                to_send.index = mlast_index;
-                self.send(to_send);
-            }
-            None => {
-                debug!(
-                    self.logger,
-                    "rejected msgApp [logterm: {msg_log_term}, index: {msg_index}] \
-                     from {from}",
-                    msg_log_term = m.log_term,
-                    msg_index = m.index,
-                    from = m.from;
-                    "index" => m.index,
-                    "tag" => &self.tag,
-                    "logterm" => ?self.raft_log.term(m.index),
-                );
-                to_send.index = m.index;
-                to_send.reject = true;
-                to_send.reject_hint = self.raft_log.last_index();
-                self.send(to_send);
+            if !append_ents.is_empty() {
```

Comment:
Why additional recovery is necessary now while unnecessary in the past?

Reply:
In this PR configration changes become active on received, so some uncommitted configuration changes could be overwritten. So here we need to do such a recover. However in the past configuration changes become active on committed, so it's unnecessary.